diesel/src/sqlite/types/date_and_time/mod.rs

```diff
@@ -92,3 +92,265 @@ impl ToSql<sql_types::TimestamptzSqlite, Sqlite> for String {
         <str as ToSql<sql_types::TimestamptzSqlite, Sqlite>>::to_sql(self as &str, out)
     }
 }
+
+#[cfg(all(test, feature = "chrono", feature = "time"))]
+mod tests {
+    extern crate chrono;
+    extern crate time;
+
+    use chrono::{DateTime, FixedOffset, NaiveDate, NaiveDateTime, NaiveTime, Utc};
+    use time::{
+        macros::{date, datetime, offset, time},
+        Date, OffsetDateTime, PrimitiveDateTime, Time,
+    };
+
+    use crate::insert_into;
+    use crate::prelude::*;
+    use crate::test_helpers::connection;
+
+    crate::table! {
+        table_timestamp_tz(id) {
+            id -> Integer,
+            timestamp_with_tz -> TimestamptzSqlite,
+        }
+    }
+    crate::table! {
+      table_timestamp(id) {
+          id -> Integer,
+          timestamp -> Timestamp
+      }
+    }
+    crate::table! {
+        table_date(id) {
+            id -> Integer,
+            date -> Date
+        }
+    }
+    crate::table! {
+        table_time(id) {
+            id -> Integer,
+            time -> Time
+        }
+    }
+
+    fn create_tables(conn: &mut SqliteConnection) {
+        crate::sql_query(
+            "CREATE TABLE table_timestamp_tz(id INTEGER PRIMARY KEY, timestamp_with_tz TEXT);",
+        )
+        .execute(conn)
+        .unwrap();
+
+        crate::sql_query("CREATE TABLE table_timestamp(id INTEGER PRIMARY KEY, timestamp TEXT);")
+            .execute(conn)
+            .unwrap();
+
+        crate::sql_query("CREATE TABLE table_date(id INTEGER PRIMARY KEY, date TEXT);")
+            .execute(conn)
+            .unwrap();
+
+        crate::sql_query("CREATE TABLE table_time(id INTEGER PRIMARY KEY, time TEXT);")
+            .execute(conn)
+            .unwrap();
+    }
+
+    #[test]
+    fn time_to_chrono_date() {
```

Comment:
Can we actually check the results of that queries for whether or not they return the "right" values?